### ISSUE TITLE
Crash in eco in non-deep mode when USE_MULTI_THREAD in use

### ISSUE THREAD

**Issue by isgursoy** [rockkingjy/OpenTracker#44]
![threaded](https://user-images.githubusercontent.com/2170245/50174347-0e3e5380-030b-11e9-962c-ac769268d581.png)

**Comment by rockkingjy** [comment]
@isgursoy yes, there is a issue here, not have time yet to debug it, if you find the issue's reason, please inform me, thanks.

**Comment by isgursoy** [comment]
it was not crashing in debug mode last time, I will print something next time.
